packages/components/forms/stories/TextInput.stories.tsx

```diff
@@ -0,0 +1,245 @@
+import React from 'react';
+
+import { TextInput, TextInputProps } from '../src';
+import { TextLink } from '@contentful/f36-text-link';
+import { LockIcon } from '@contentful/f36-icons';
+import { SectionHeading } from '@contentful/f36-typography';
+import { Flex } from '@contentful/f36-core';
+import { SearchIcon } from '@contentful/f36-icons';
+
+export default {
+  title: 'Form Elements/TextInput',
+  component: TextInput,
+  parameters: {
+    propTypes: [TextInput['__docgenInfo']],
+  },
+  args: {
+    onBlur: undefined,
+    onChange: undefined,
+    onFocus: undefined,
+  },
+  argTypes: {
+    className: { control: { disable: true } },
+    testId: { control: { disable: true } },
+    style: { control: { disable: true } },
+  },
+};
+
+export const Basic = (args: TextInputProps) => {
+  return (
+    <TextInput
+      {...args}
+      name="Example name"
+      value="Example value"
+      id="input-1"
+      link={
+        <TextLink icon={<LockIcon />} href="https://f36.contentful.com/">
+          go to link
+        </TextLink>
+      }
+    />
+  );
+};
+
+Basic.args = {
+  label: 'Example label for input',
+  validationMessage: 'validation message',
+  helpText: 'help text',
+  placeholder: 'this is my placeholder',
+};
```

Comment:
IMO the Basic story should be the most, well, basic thing you can do. In this case, I think we should pass only label and placeholder

Then maybe we can create a story showing how to handle errors and that would use `validationMessage`

One showing the helpText thing as well